updating to aeson, negotiating [From,To]JSON instances

Anton-Latukha opened the pull request:
I tried to use new `aeson` in `default.nix` - that required ~5 more overrides and resulted in infinite recursion, so I decided to stick to main agenda - follow Hackage and sync with `aeson`, and approach Nixpkgs infrastructure after.

sjakobi reviewed (approved):
:+1: for the library change. Cheers!

Anton-Latukha committed:
cabal: aeson < 1.5.3

1.5.3 introduced instances for `strict` & `data-fix`, some intstances overlap.

Trying to remove/sync instances results in:

```text
src/Nix/Expr/Types.hs:1:1: error:
    Exception when trying to run compile-time code:
      Constructor ‘DynamicKey‘ must only use its last 1 type variable(s) within
      the last 1 argument(s) of a data type
```
